src/prover_server.cc

```diff
@@ -90,9 +89,9 @@ class ProverImpl final : public Prover::Service {
 
     // Parse received message to feed to the prover
     try {
-      libzeth::bits256 root_bits = libzeth::hexadecimal_digest_to_bits256(proofInputs->root());
-      libzeth::bits64 vpub_in = libzeth::hexadecimal_value_to_bits64(proofInputs->inpubvalue());
-      libzeth::bits64 vpub_out = libzeth::hexadecimal_value_to_bits64(proofInputs->outpubvalue());
+      FieldT root = string_to_field<FieldT>(proofInputs->root());
```

Comment:
string_to_field does recognize the string as hex ?

Reply:
Yes